src/coreComponents/physicsSolvers/multiphysics/ThermalSinglePhasePoromechanicsFixedStressSolver.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file ThermalSinglePhasePoromechanicsFixedStressSolver.cpp
+ *
+ */
+
+#include "ThermalSinglePhasePoromechanicsFixedStressSolver.hpp"
+
+#include "constitutive/ConstitutiveManager.hpp"
+#include "discretizationMethods/NumericalMethodsManager.hpp"
+#include "mesh/DomainPartition.hpp"
+
+namespace geosx
+{
+
+using namespace dataRepository;
+using namespace constitutive;
+
+ThermalSinglePhasePoromechanicsFixedStressSolver::ThermalSinglePhasePoromechanicsFixedStressSolver( const string & name,
+                                                                                                    Group * const parent ):
+  Base( name, parent )
+{}
+
+void ThermalSinglePhasePoromechanicsFixedStressSolver::initializePreSubGroups()
+{
+  SolverBase::initializePreSubGroups();
+
+  // Set m_updateThermalPoroElasticityFlag in SolidMechanicsLagrangianFEM to 1
+  solidMechanicsSolver()->turnOnFixedStressThermoPoroElasticityFlag();
+}
+
+void ThermalSinglePhasePoromechanicsFixedStressSolver::postProcessInput()
+{
+  Base::postProcessInput();
+  GEOSX_WARNING_IF( getNonlinearSolverParameters().m_couplingType == NonlinearSolverParameters::CouplingType::FullyImplicit,
+                    "FullyImplicit coupling not implemented for this solver. A sequential coupling approach will be used." );
+  getNonlinearSolverParameters().m_couplingType = NonlinearSolverParameters::CouplingType::Sequential;
+}
+
+ThermalSinglePhasePoromechanicsFixedStressSolver::~ThermalSinglePhasePoromechanicsFixedStressSolver()
+{
+  // TODO Auto-generated destructor stub
+}
+
+void ThermalSinglePhasePoromechanicsFixedStressSolver::mapSolutionBetweenSolvers( DomainPartition & domain, integer const solverType )
```

Comment:
I think that if you move this function to `SinglePhasePoromechanics` we don't need this solver but everything should work with `SinglePhasePoromechanics` if `couplingType = "Sequential"`